src/lib/snack-bar/snack-bar.spec.ts

```diff
@@ -0,0 +1,159 @@
+import {
+  inject,
+  async,
+  ComponentFixture,
+  TestBed
+} from '@angular/core/testing';
+import {
+  NgModule,
+  Component,
+  Directive,
+  ViewChild,
+  ViewContainerRef
+} from '@angular/core';
+import {MdSnackBar, MdSnackBarModule} from './snack-bar';
+import {OverlayContainer} from '../core';
+import {MdSnackBarConfig} from './snack-bar-config';
+import {SimpleSnackBar} from './simple-snack-bar';
+
+
+describe('MdSnackBar', () => {
+  let snackBar: MdSnackBar;
+  let overlayContainerElement: HTMLElement;
+
+  let testViewContainerRef: ViewContainerRef;
+  let viewContainerFixture: ComponentFixture<ComponentWithChildViewContainer>;
+
+  let simpleMessage = 'Burritos are here!';
+  let simpleActionLabel = 'pickup';
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [MdSnackBarModule.forRoot(), SnackBarTestModule],
+      providers: [
+        {provide: OverlayContainer, useFactory: () => {
+          overlayContainerElement = document.createElement('div');
+          return {getContainerElement: () => overlayContainerElement};
+        }}
+      ],
+    });
+    TestBed.compileComponents();
+  }));
+
+  beforeEach(inject([MdSnackBar], (sb: MdSnackBar) => {
+    snackBar = sb;
+  }));
+
+  beforeEach(() => {
+    viewContainerFixture = TestBed.createComponent(ComponentWithChildViewContainer);
+
+    viewContainerFixture.detectChanges();
+    testViewContainerRef = viewContainerFixture.componentInstance.childViewContainer;
+  });
+
+  it('should have the role of alert', () => {
+    let config = new MdSnackBarConfig(testViewContainerRef);
+    snackBar.open(simpleMessage, simpleActionLabel, config);
+
+    let containerElement = overlayContainerElement.querySelector('snack-bar-container');
+
+    expect(containerElement.getAttribute('role')).toBe('alert');
```

Comment:
One new thing I want to start doing is adding a message to asserts. This one would be something like
```ts
expect(containerElement.getAttribute('role'))
    .toBe('alert', 'Expected snackbar container to have role="alert"');
```

I only found out you could do this last week, and it makes test failures much easier to quickly understand.